### DISCUSSION TITLE
`set_expiry` not working as expected

### DISCUSSION THREAD

**Discussion by johnbchron** [maxcountryman/axum-login#194] [General]
`set_expiry()` isn't working as I expected; if I set it to something like `Expiry::OnInactivity(time::Duration::seconds(15))`, it doesn't expire the session. Calling `save()` doesn't seem to do anything.

When I inspect the internals of `Session`, it seems that the `expiry` field is getting set correctly, but `record.expiry_date` never changes. Am I using this correctly?

**Comment by maxcountryman** [comment]
Can you please provide a minimal example that demonstrates what you're seeing?

Feel free to augment one of the examples and share it with us so we can take a look.

**Comment by johnbchron** [comment]
Will do; I'm travelling right now but I'll do this in a couple days.

**Comment by maxcountryman** [comment]
I believe we've addressed this in the latest `tower-sessions` release: this is available as version `0.11.1` which fixes a bug where the underlying session record expiry was not being updated after initially setting it.
